### ISSUE TITLE
Continue training after initial training gracefully ends

### ISSUE THREAD

**Issue by dcapeluto** [mljar/mljar-supervised#568]
It is hard to know an adequate total_time_limit for a specific scenario in AutoML. The time limit depends on data size, training machine capability and algorithms chosen. Selecting a seemingly outrageously high limit, could end up cutting the training short and skipping algorithms, while AutoML moves on to Ensemble/Stack tasks with limited set of trained algorithms, due to this limit. Obviously this can be worked around by retraining from scratch with a higher total_time_limit, but if there were several hours (e.g. 24) already trained, those cannot be used towards continuing the training.

As such, if there were a feature to "continue training" after training gracefully ends, and pick up where it left off and continue training the skipped algorithms, and then re-training the existing ensemble/stacking/etc, but taking into account the newly trained algos skipped in the first training job, would be very useful. If there's a manual way of achieving this (I suspect tweaking params.json, progress.json and existing folders), and anyone knows how's done, I could potentially write a method that makes the adjustments to these, and commit the code for review and merge. 

PS: Overall, amazing AutoML library. Very easy to use and highly recommended from any "citizen data scientist" looking to get useful results without acquiring the degree first... ;)

**Comment by dcapeluto** [comment]
@AkshayNovacene no unfortunately. I did a little bit of trying to mimic the internal files to simulate an interruption in training so that it would continue instead of thinking that it finished, but couldn't get it to work. I had to restart every time.

**Comment by AkshayNovacene** [comment]
> @AkshayNovacene no unfortunately. I did a little bit of trying to mimic the internal files to simulate an interruption in training so that it would continue instead of thinking that it finished, but couldn't get it to work. I had to restart every time.

Do you think there's a workaround for this?

**Comment by dcapeluto** [comment]
Yes. See the files created while training. They contain the status and progress of the training. You can probably mimic training interruption as opposed to training completion and when you relaunch the training, the program would then "see" that the training was interrupted and pick up where it left off, assuming you also increased the training time. @pplonski probably would know off the top of his head how to do this.

**Comment by pplonski** [comment]
Hi @AkshayNovacene,

As @dcapeluto wrote, you can try to mimic the training interruption with longer training time. Maybe there will be required change in tuner to generate more unique configurations of hyper-parameters.
